src/App.tsx

```diff
@@ -1,61 +1,136 @@
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [titleHasError, setTitleHasError] = useState(false);
+  const [userHasError, setUserHasError] = useState(false);
+  const [newTodos, setNewTodos] = useState(todos);
+
+  const addTodo = () => {
+    if (userId && title) {
+      const newId = Math.max(...newTodos.map(todo => todo.id)) + 1;
+
+      const newTodo = {
+        id: newId,
+        userId,
+        completed: false,
+        title,
+        user: getUser(userId),
+      };
+
+      setNewTodos((currentTodos) => ([...currentTodos, newTodo]));
+
+      setUserId(0);
+      setTitle('');
+      setUserHasError(false);
+      setTitleHasError(false);
+    }
+
+    if (!userId) {
+      setUserHasError(true);
+    }
+
+    if (!title) {
+      setTitleHasError(true);
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    addTodo();
+  };
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleHasError(false);
+  };
+
+  const handleUserId = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserHasError(false);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={title}
+              onChange={handleTitle}
+            />
+          </label>
+
+          {titleHasError && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label>
+            <select
+              data-cy="userSelect"
+              name="user"
+              value={userId}
+              onChange={handleUserId}
+            >
+              <option
+                value="0"
+                disabled
+              >
+                Choose a user
+              </option>
+              {usersFromServer.map(user => (
+                <option
+                  key={user.id}
+                  value={user.id}
+                >
+                  {user.name}
+                </option>
+              ))}
```

Comment:
object destructuring 
```suggestion
              {usersFromServer.map(({id, name}) => (
                <option
                  key={id}
                  value={id}
                >
                  {name}
                </option>
              ))}
```